asset_tracker_app/lib/services/firebase/auth_service.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:firebase_auth/firebase_auth.dart';
+
+class AuthService {
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+
+  // Sign in.
+  Future<bool> signIn(String email, String password) async {
+    try {
+      await _auth.signInWithEmailAndPassword(
+        email: email,
+        password: password,
+      );
+      return true;
+    } on FirebaseAuthException catch (e) {
+      throw _handleAuthError2(e);
+    } catch (e) {
+      if (_auth.currentUser != null) {
+        return true;
+      }
+      throw "There was an error logging in. Please try again.";
+    }
+  }
+
+  String _handleAuthError2(FirebaseAuthException e) {
+    switch (e.code) {
+      case 'user-not-found':
```

Comment:
buradaki ifadeler bir enum ile tanımlanabilir. extension ile string'ten enum'a çevrilebilir. mesajlar için yukarıdaki yorumum geçerli